tests/iteminfo_test.cpp

```diff
@@ -17,88 +19,294 @@ static void iteminfo_test( const item &i, const iteminfo_query &q, const std::st
+                        iteminfo_parts::ARMOR_COVERAGE, iteminfo_parts::ARMOR_WARMTH,
+                        iteminfo_parts::ARMOR_ENCUMBRANCE, iteminfo_parts::ARMOR_PROTECTION
                       } );
-    iteminfo_test(
-        item( "compbow" ), q,
-        "--\n"
-        "Damage: <color_c_yellow>18</color>*<color_c_yellow>1.25</color> = <color_c_yellow>22</color>\n" );
+
+    SECTION( "shows coverage, encumbrance, and protection for armor with coverage" ) {
+        iteminfo_test(
+            item( "longshirt" ), q,
+            "--\n"
+            // NOLINTNEXTLINE(cata-text-style)
+            "Covers: The <color_c_cyan>torso</color>. The <color_c_cyan>arms</color>. \n"
+            // NOLINTNEXTLINE(cata-text-style)
+            "Layer: <color_c_light_blue>Normal</color>. \n"
+            "Coverage: <color_c_yellow>90</color>%  Warmth: <color_c_yellow>5</color>\n"
+            "--\n"
+            "<color_c_white>Encumbrance</color>: <color_c_yellow>3</color> <color_c_red>(poor fit)</color>\n"
+            "<color_c_white>Protection</color>: Bash: <color_c_yellow>1</color>  Cut: <color_c_yellow>1</color>\n"
+            "  Acid: <color_c_yellow>0</color>  Fire: <color_c_yellow>0</color>  Environmental: <color_c_yellow>0</color>\n" );
+    }
+
+    SECTION( "omits irrelevant info if it covers nothing" ) {
+        iteminfo_test(
+            item( "ear_plugs" ), q,
+            "--\n"
+            "Covers: <color_c_cyan>Nothing</color>.\n" );
+    }
 }
 
-TEST_CASE( "nutrients_in_regular_item", "[item][iteminfo]" )
+TEST_CASE( "ranged weapon attributes", "[item][iteminfo]" )
+{
+    SECTION( "ammo capacity" ) {
+        iteminfo_query q( { iteminfo_parts::GUN_CAPACITY } );
+        iteminfo_test(
+            item( "compbow" ), q,
+            "--\n"
+            "<color_c_white>Capacity:</color> <color_c_yellow>1</color> round of arrows\n" );
+    }
+
+    SECTION( "default ammo when unloaded" ) {
+        iteminfo_query q( { iteminfo_parts::GUN_DEFAULT_AMMO } );
+        iteminfo_test(
+            item( "compbow" ), q,
+            "--\n"
+            "Gun is not loaded, so stats below assume the default ammo:"
+            " <color_c_light_blue>wooden broadhead arrow</color>\n" );
+    }
+
+    SECTION( "damage including floating-point multiplier" ) {
+        iteminfo_query q( { iteminfo_parts::GUN_DAMAGE, iteminfo_parts::GUN_DAMAGE_AMMOPROP,
+                            iteminfo_parts::GUN_DAMAGE_TOTAL
+                          } );
+        iteminfo_test(
+            item( "compbow" ), q,
+            "--\n"
+            "Damage: <color_c_yellow>18</color>*<color_c_yellow>1.25</color> = <color_c_yellow>22</color>\n" );
+    }
+
+    SECTION( "time to reload" ) {
+        iteminfo_query q( { iteminfo_parts::GUN_RELOAD_TIME } );
+        iteminfo_test(
+            item( "compbow" ), q,
+            "--\n"
+            "Reload time: <color_c_yellow>110</color> moves\n" );
```

Comment:
I seem to have given bad advice, lets disable the lint for this line.
```suggestion
            "Reload time: <color_c_yellow>110</color> moves \n" ); // NOLINT(cata-text-style)
```

Reply:
Good call. Thanks for reviewing!